packages/builders/src/messages/poll/Poll.ts

```diff
@@ -0,0 +1,204 @@
+	 * 		{ text: 'Answer 1' },
+	 * 		{ text: 'Answer 2' },
+	 * 	);
+	 * ```
+	 * @param answers - The answers to add
+	 */
+	public addAnswers(...answers: RestOrArray<APIPollMedia>): this {
+		const normalizedAnswers = normalizeArray(answers);
+
+		// Ensure adding these answers won't exceed the 10 answer limit
+		validateAnswerLength(normalizedAnswers.length, this.data.answers);
+
+		// Data assertions
+		pollAnswersArrayPredicate.parse(normalizedAnswers);
+
+		if (this.data.answers) {
+			this.data.answers.push(...normalizedAnswers.map((answer) => ({ poll_media: answer })));
+		} else {
+			this.data.answers = normalizedAnswers.map((answer) => ({ poll_media: answer }));
+		}
+
+		return this;
+	}
+
+	/**
+	 * Removes, replaces, or inserts answers for this poll.
+	 *
+	 * @remarks
+	 * This method behaves similarly
+	 * to {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice | Array.prototype.splice()}.
+	 * The maximum amount of answers that can be added is 10.
+	 *
+	 * It's useful for modifying and adjusting order of the already-existing answers of a poll.
+	 * @example
+	 * Remove the first answer:
+	 * ```ts
+	 * poll.spliceAnswers(0, 1);
+	 * ```
+	 * @example
+	 * Remove the first n answers:
+	 * ```ts
+	 * const n = 4;
+	 * poll.spliceAnswers(0, n);
+	 * ```
+	 * @example
+	 * Remove the last answer:
+	 * ```ts
+	 * poll.spliceAnswers(-1, 1);
+	 * ```
+	 * @param index - The index to start at
+	 * @param deleteCount - The number of answers to remove
+	 * @param answers - The replacing answer objects
+	 */
+	public spliceAnswers(index: number, deleteCount: number, ...answers: RestOrArray<APIPollMedia>): this {
+		const normalizedAnswers = normalizeArray(answers);
+
+		// Ensure adding these answers won't exceed the 10 answer limit
+		validateAnswerLength(answers.length - deleteCount, this.data.answers);
+
+		// Data assertions
+		pollAnswersArrayPredicate.parse(answers);
+
+		if (this.data.answers) {
+			this.data.answers.splice(index, deleteCount, ...normalizedAnswers.map((answer) => ({ poll_media: answer })));
+		} else {
+			this.data.answers = normalizedAnswers.map((answer) => ({ poll_media: answer }));
+		}
+
+		return this;
+	}
+
+	/**
+	 * Sets the answers for this poll.
+	 *
+	 * @remarks
+	 * This method is an alias for {@link PollBuilder.spliceAnswers}. More specifically,
+	 * it splices the entire array of answers, replacing them with the provided answers.
+	 *
+	 * You can set a maximum of 10 answers.
+	 * @param answers - The answers to set
+	 */
+	public setAnswers(...answers: RestOrArray<APIPollMedia>): this {
+		this.spliceAnswers(0, this.data.answers?.length ?? 0, ...normalizeArray(answers));
+		return this;
+	}
+
+	/**
+	 * Sets the question for this poll.
+	 *
+	 * @param text - The question to use
+	 */
+	public setQuestion(text: string): this {
+		// Data assertions
+		pollQuestionPredicate.parse(text);
+
+		this.data.question = { text };
+		return this;
+	}
```

Comment:
Discord states that this only accepts text "for now". [^1] Allowing only a string isn't particularly future-proof. Would rather want an object rather than more parameters.

[^1]: https://discord.com/developers/docs/resources/poll#poll-media-object